playground/knot/src/Editor/Editor.tsx

```diff
@@ -156,9 +159,68 @@ class PlaygroundServer
       "python",
       this,
     );
+    this.inlayHintsDisposable = monaco.languages.registerInlayHintsProvider(
+      "python",
+      this,
+    );
     this.editorOpenerDisposable = monaco.editor.registerEditorOpener(this);
   }
 
+  provideInlayHints(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _model: editor.ITextModel,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _range: Range,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _token: CancellationToken,
+  ): languages.ProviderResult<languages.InlayHintList> {
+    const workspace = this.props.workspace;
+    const selectedFile = this.props.files.selected;
+
+    if (selectedFile == null) {
+      return;
+    }
+
+    const selectedHandle = this.props.files.handles[selectedFile];
+
+    if (selectedHandle == null) {
+      return;
+    }
+
+    const inlayHints = workspace.inlayHints(selectedHandle);
+
+    if (inlayHints.length === 0) {
+      return undefined;
+    }
+
+    return {
+      dispose: () => {},
+      hints: inlayHints.map(
+        (hint: {
+          position: { line: number; column: number };
+          markdown: string;
+        }) => {
+          return {
```

Comment:
Annotations like these in a lambda are uncommon. It's more common to annotate the variable `inlayHints`. But my preference here is to rely on TypeScript's inference
```suggestion
        hint => {
          return {
```